fakestorage/xml_response.go

```diff
@@ -12,6 +12,14 @@ type xmlResponse struct {
 	errorMessage string
 }
 
+type xmlResponseBody struct {
+	XMLName  xml.Name `xml:"PostResponse"`
+	Bucket   string
+	Etag     interface{}
```

Comment:
Any reason to use interface{} here instead of the struct declared below?

Reply:
Not particularly. I get confused with how Go handles nested structs and their assignability, but have adjusted this now. Let me know if you have any comments or a better way of doing it :).